src/framework/theme/modal/modalResolver.component.tsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import React from 'react';
+import {
+  View,
+  ViewProps,
+  StyleSheet,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+
+type ChildElement = React.ReactElement<any>;
+type ChildrenProp = ChildElement | ChildElement[];
+
+interface ComponentProps {
+  visible: boolean;
+  children: ChildrenProp;
+  isBackDropAllowed?: boolean;
+  identifier?: string;
+  onCloseModal?: (index: string) => void;
+}
+
+export type ModalResolverProps = ViewProps & ComponentProps;
+
+export class ModalResolver extends React.Component<ModalResolverProps> {
+
+  static defaultProps: Partial<ModalResolverProps> = {
+    visible: false,
+    isBackDropAllowed: false,
+  };
+
+  private closeModal: () => void = (): void => {
+    if (this.props.onCloseModal) {
+      this.props.onCloseModal(this.props.identifier);
+    }
+  };
+
+  private closeOnBackdrop: () => void = () => {
+    if (this.props.isBackDropAllowed) {
+      this.closeModal();
+    }
+  };
+
+  private onStartShouldSetResponder = (): boolean => true;
+
+  private onResponderRelease = (): void => {
+    return;
+  };
+
+  private onStartShouldSetResponderCapture = (): boolean => false;
+
+  private renderComponentChild = (source: React.ReactElement<any>): React.ReactElement<any> => {
+    return React.cloneElement(source, {
+      ...source.props,
+      onCloseModal: this.closeModal,
+      style: {
+        ...source.props.style,
+        ...StyleSheet.flatten(this.props.style),
+      },
+    });
+  };
+
+  private renderComponentChildren = (source: React.ReactNode): React.ReactElement<any>[] => {
+    return React.Children.map(source, this.renderComponentChild);
+  };
+
+  private renderWithBackDrop = (component: React.ReactElement<ViewProps>) => (
```

Comment:
prefer using both brackets and return statement:
```
renderAwesomeComponent = (): React.ReactElement<AwesomeComponentProps> => {
  return (
    <Awesome/>
  );
};
```

Reply:
Done.